service_management/app/repositories/analysis/dao/AnalysisDao.scala

```diff
@@ -131,6 +146,42 @@ class AnalysisDao @Inject()(
   }
 
   /**
+   * Performs an update action against the DB using the values in the provided
+   * {{{AnalysisEvent}}} argument.
+   *
+   * @param id the EventId associated with the analysis event to update
+   * @param ae the AnalysisEvent to update
+   * @return a result with an option of the updated event
+   */
+  def update(
+      id: EventId,
+      ae: AnalysisEvent
+  ): Future[MusitResult[Option[AnalysisEvent]]] = {
+    val action = updateAction(id, ae).transactionally
```

Comment:
I prefer that we change name of the `updateAction` function to indicate that it update by the `id`. Also I think that will make the flatmap code below a bit easer to understand sine we can/should only have zero or one updated row. The reason for it is that I thought about the case where it was higher than 1 and if we should rollback the update, something that should  never happen.